libs/muddle/src/muddle_register.cpp

```diff
@@ -287,7 +287,7 @@ void MuddleRegister::Enter(WeakConnectionPtr const &ptr)
  */
 void MuddleRegister::Leave(ConnectionHandle handle)
 {
-  FETCH_LOCK(lock_);
+  lock_.lock();
```

Comment:
As an alternative, this can be done by enclosing `lock_.lock(); ... lock_.unlock();` part in braces, keeping `FETCH_LOCK` intact.

Reply:
You could do this, I think it seems cleaner not to, since you would need to declare the callback_copy outside the locked scope, then assign it inside the locked scope for use outside the locked scope right?

Reply:
Sure, but you will also have the guarantee lock is released no matter what, should you need one.
Btw, an alternative would be to design a custom guard class that does the same acquire on ctor/release on destruction but also has an explicit method for the latter.

Reply:
You're right, I will push this change